javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/model/typesystem/LazyType.java

```diff
@@ -80,20 +52,32 @@ public LazyType(Supplier<ResolvedType> supplier) {
         this.supplier = supplier;
     }
 
-    private ResolvedType getType() {
+    /**
+     * Get the type wrapped by a {@link LazyType}.
+     *
+     * This method is used to get and unpack the {@link LazyType} when they are nested.
+     *
+     * {@code
+     *      LazyType typeA = new LazyType(() -> ResolvedVoidType.INSTANCE);
+     *      LazyType typeB = new LazyType(() -> typeA);
+     *      LazyType typeC = new LazyType(() -> typeB);
+     * }
+     *
+     * In this examples, by calling {@code getType()} we would expected all of them to return {@link ResolvedVoidType}
+     *
+     * @return The concrete type for the {@link LazyType}.
+     */
+    public ResolvedType getType() {
```

Comment:
Why are you exposing the underlying type? The LazyType acts as a proxy and delegates behavior to the wrapped type. Is there a special need to do this?

Reply:
Essentially just for testing purposes.  Maybe we can reduce the visibility to package instead of public?

Reply:
If we consider that the scope of this method is private (it is not useful to test it directly). Isn't invoking the equals method enough to test the behavior of the getType method?